src/Security/Authentication/WsFederation/src/WsFederationHandler.cs

```diff
@@ -195,7 +195,7 @@ protected override async Task<HandleRequestResult> HandleRemoteAuthenticateAsync
             {
                 // Extract the user state from properties and reset.
                 properties.Items.TryGetValue(WsFederationDefaults.UserstatePropertiesKey, out var userState);
-                wsFederationMessage.Wctx = userState;
+                wsFederationMessage.Wctx = userState!;
```

Comment:
This looks like a bug in IdentityModel. `wctx` is OPTIONAL according to http://docs.oasis-open.org/wsfed/federation/v1.2/os/ws-federation-1.2-spec-os.html#_Toc223175023.

@jennyf19 @brentschmaltz - can you open an issue in https://github.com/azuread/azure-activedirectory-identitymodel-extensions-for-dotnet/ to track annotating these properties correctly for `nullable`? Then link to the bug here.

Reply:
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2240